### PR TITLE
Correct BlockPublisher test for empty blocks to use the CandidateBloc…

### PR DESCRIPTION
**Pull request by cianx**
…k state.

Signed-off-by: Cian Montgomery <cian.montgomery@intel.com>

### CODE REVIEW

**Comment by cianx** [comment]
Addresses the root cause of this issue, 
```
```
[19:07:46.282 CRITICAL publisher] on_check_publish_block exception.                                                   
[19:07:46.282 ERROR    publisher] None is not in list                                                                 
Traceback (most recent call last):                                                                                    
  File "/project/sawtooth-core/validator/sawtooth_validator/journal/publisher.py", line 505, in on_check_publish_block
    last_batch_index = self._pending_batches.index(last_batch)                                                        
ValueError: None is not in list  
```
Originally seen by @peterschwarz

File: validator/sawtooth_validator/journal/publisher.py [hunk 1]
```diff
@@ -490,8 +494,9 @@ def on_check_publish_block(self, force=False):
                         self._pending_batches:
                     self._build_candidate_block(self._chain_head)
 
-                if self._candidate_block and \
-                        (force or self._pending_batches) and \
+                if self._candidate_block and (
+                            force or
+                            self._candidate_block.has_pending_batches()) and \
```
Comment (peterschwarz) [inline validator/sawtooth_validator/journal/publisher.py]: Given that if `_pending_batches` is empty, it evaluates to false, this doesn't necessarily fix the cause as to why `last_batch` was being called in the first place.  The error message from last_batch might be slightly more informative, though.

Comment (cianx) [inline validator/sawtooth_validator/journal/publisher.py]: self._pending_batches and self._candidate_block._pending_batches are different lists and are not guarenteed to have the same content. self._candidate_block.last_batch should never be called on a Candididate block as we don't publish empty blocks.

Comment (peterschwarz) [inline validator/sawtooth_validator/journal/publisher.py]: The diff did not make that clear.   I think I see now.

Comment (peterschwarz) [inline validator/sawtooth_validator/journal/publisher.py]: A more informative commit message would have made it easier to understand what the change was.

**Pull request merged**